wasm-runtime/polkadot/src/codec/slicable.rs

```diff
@@ -43,12 +44,14 @@ pub trait Slicable: Sized {
 /// Trait to mark that a type is not trivially (essentially "in place") serialisable.
 pub trait NonTrivialSlicable: Slicable {}
 
-impl<T: EndianSensitive> Slicable for T {
-	fn set_as_slice<F: FnOnce(&mut[u8]) -> bool>(fill_slice: F) -> Option<Self> {
-		let size = size_of::<T>();
-		let mut result: T = unsafe { uninitialized() };
+// note: the copy bound and static lifetimes are necessary for safety of `set_as_slice`.
+impl<T: Copy + EndianSensitive + 'static> Slicable for T {
+	fn set_as_slice<F: FnOnce(&mut [u8]) -> bool>(fill_slice: F) -> Option<Self> {
+		let size = mem::size_of::<T>();
+		let mut result: T = unsafe { mem::zeroed() };
```

Comment:
why is this better? it's obvious that it'll be written to; looks like it's just an invitation for the compiler's optimiser to fail.

Reply:
reading from zeroed memory is allowed. reading from uninitialized memory is undefined behavior. "reading" in this case includes writing without using `ptr::write`, which the `fill_slice` closure is probably expected to do.

Reply:
well, at least it'll likely be in L0 cache when the zeroes are being overwritten with the real data.